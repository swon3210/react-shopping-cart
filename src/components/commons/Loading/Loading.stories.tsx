import Loading from './Loading';

export default {
  title: 'Components/Commons/Loading',
  component: Loading,
  argTypes: {},
};

const Template = () => <Loading />;

export const Default = Template.bind({});
