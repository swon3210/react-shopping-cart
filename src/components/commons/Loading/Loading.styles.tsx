import styled from 'styled-components';

export const Loading = styled.div`
  width: 100%;
  height: ${({ theme }) => theme.PAGE_HEIGHT};
  display: flex;
  justify-content: center;
  align-items: center;
`;

export const Image = styled.img`
  transform: translateY(-50%);
`;
