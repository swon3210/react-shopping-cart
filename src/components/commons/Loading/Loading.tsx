import * as Styled from './Loading.styles';
import loadingSVG from '../../../assets/svgs/loading.svg';

const Loading = () => {
  return (
    <Styled.Loading>
      <Styled.Image src={loadingSVG} alt="로딩 중"></Styled.Image>
    </Styled.Loading>
  );
};

export default Loading;
