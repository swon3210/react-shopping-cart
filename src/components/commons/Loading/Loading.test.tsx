import Loading from './Loading';
import { render } from '@testing-library/react';

describe('Loading Component', () => {
  it('Loading Snapshot', () => {
    const LoadingUtil = render(<Loading />);
    expect(LoadingUtil.asFragment()).toMatchSnapshot();
  });
});
