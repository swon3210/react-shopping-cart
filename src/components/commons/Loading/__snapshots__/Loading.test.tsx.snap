// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Loading Component Loading Snapshot 1`] = `
<DocumentFragment>
  <div
    class="sc-bdvvaa gmpVqK"
  >
    <img
      alt="로딩 중"
      class="sc-gsDJrp dVraHg"
      src="loading.svg"
    />
  </div>
</DocumentFragment>
`;
